src/rawtoaces_util/image_converter.cpp

```diff
@@ -113,31 +130,25 @@ std::vector<std::string> database_paths()
         path = default_path;
     }
 
-    size_t pos = 0;
-
-    while ( pos < path.size() )
-    {
-        size_t end = path.find( separator, pos );
-
-        if ( end == std::string::npos )
-            end = path.size();
-
-        std::string pathItem = path.substr( pos, end - pos );
-
-        if ( find( result.begin(), result.end(), pathItem ) == result.end() )
-            result.push_back( pathItem );
-
-        pos = end + 1;
-    }
+    OIIO::Strutil::split( path, result, separator );
 
     return result;
 };
 
-bool fetch_camera_make_and_model(
-    const OIIO::ImageSpec &spec,
-    std::string           &camera_make,
-    std::string           &camera_model )
+/// Get camera info (with make and model) from image metadata or custom settings.
+///
+/// Returns camera information using custom settings if provided, otherwise
+/// extracts from image metadata. Returns empty CameraInfo if required metadata is missing.
+///
+/// @param spec Image specification containing metadata
+/// @param settings Converter settings with optional custom camera info
+/// @return CameraInfo struct with make and model, or empty if unavailable
+rta::core::CameraInfo get_camera_info(
+    const OIIO::ImageSpec &spec, const ImageConverter::Settings &settings )
```

Comment:
a bit more functional - you give sources of where to get camera info from, you get camera info out.